src/zkevm_specs/evm/execution/codecopy.py

```diff
@@ -0,0 +1,51 @@
+from ...util import N_BYTES_MEMORY_ADDRESS, FQ
+from ..execution_state import ExecutionState
+from ..instruction import Instruction, Transition
+from ..step import CopyCodeToMemoryAuxData
+from ..table import RW, RWTableTag, CallContextFieldTag, AccountFieldTag
+
+
+def codecopy(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    memory_offset_word, code_offset_word, size_word = (
+        instruction.stack_pop(),
+        instruction.stack_pop(),
+        instruction.stack_pop(),
+    )
+
+    memory_offset, size = instruction.memory_offset_and_length(memory_offset_word, size_word)
+    code_offset = instruction.rlc_to_fq_exact(code_offset_word, N_BYTES_MEMORY_ADDRESS)
+
+    account = instruction.call_context_lookup(CallContextFieldTag.CalleeAddress)
+    code_hash = instruction.account_read(account, AccountFieldTag.CodeHash)
+    instruction.constrain_equal(code_hash, instruction.curr.code_source)
```

Comment:
Just realized `CODECOPY` copies current executed code, including contract creation code. While contract creation, the account's code hash is not updated actually.

These lines should be removed, and we should use `code_source` as the `aux.code_hash`, it will always be the pointer to where the code is executed (as for why it's `code_source` instead of `code_hash`, see #73).
```suggestion
```